### DISCUSSION TITLE
MemoryRouter?

### DISCUSSION THREAD

**Discussion by LookRain** [molefrog/wouter#466] [Q&A]
This looks like an amazing project! However, I'm looking for something more like the <MemoryRouter/> offered by react router to manage a quite complex component states at my work. Does Wouter offer something like that?

**Comment by molefrog** [comment]
Yes, wouter uses *location hooks* to configure how navigation is done. You can write your own ones, but there are some that are available in the library, including `useMemoryLocation`. I think, this could be helpful

https://github.com/molefrog/wouter?tab=readme-ov-file#how-do-i-configure-the-router-to-render-a-specific-route-in-tests
